File: src/test/java/sds/offre/FormuleRepositoryEnMemoire.java
package sds.offre;

import sds.offre.concept_metier.Formule;
import sds.offre.concept_metier.FormuleRepository;
import sds.offre.concept_metier.IdFormule;

import java.util.Collection;
import java.util.HashMap;

public class FormuleRepositoryEnMemoire implements FormuleRepository {

    private HashMap<IdFormule, Formule> formuleMap = new HashMap<>();

    @Override
    public Formule get(IdFormule id) {
        return formuleMap.get(id);
    }

    @Override
    public void addOrReplace(Formule formule) {
        formuleMap.put(formule.getId(), formule);
    }

    @Override
    public Collection<Formule> getToutesLesFormules() {
        return formuleMap.values();
    }
}

File: src/test/java/sds/offre/FormuleRepositoryEnMemoire.kt
package sds.offre

import sds.offre.concept_metier.Formule
import sds.offre.concept_metier.FormuleRepository
import sds.offre.concept_metier.IdFormule
import java.util.*

class FormuleRepositoryEnMemoire : FormuleRepository {

    private val formuleMap = HashMap<IdFormule, Formule>()

    override operator fun get(id: IdFormule): Formule {
        return formuleMap[id] ?: throw FormuleNonTrouvéeException("")
    }

    override fun addOrReplace(formule: Formule) {
        formuleMap[formule.id] = formule
    }

    override fun toutesLesFormules(): Collection<Formule> {
        return formuleMap.values
    }
}
